src/backend/utils/adt/agtype_ops.c

```diff
@@ -1232,12 +1231,21 @@ Datum agtype_exists_agtype(PG_FUNCTION_ARGS)
         PG_RETURN_BOOL(false);
     }
 
-    str = get_string_from_agtype_value(aval, &aval->val.string.len);
-    aval = string_to_agtype_value(str);
+    if (AGT_ROOT_IS_OBJECT(agt) &&
+        aval->type == AGTV_STRING)
+    {
+        v = find_agtype_value_from_container(&agt->root,
+                                            AGT_FOBJECT,
+                                            aval);
+    }
+    else if (AGT_ROOT_IS_ARRAY(agt) &&
+                !aval->type == AGTV_NULL)
```

Comment:
Will this query crash the server given only these two conditions?
`SELECT '[1,2,3]'::agtype ? '{"id": 1688849860263937, "label": "EDGE", "end_id": 1970324836974593, "start_id": 1407374883553281, "properties": {"n": 100}}::edge'::agtype;`
And what about this one?
`SELECT '{"id": 281474976710658, "label": "", "properties": {"n": 100}}'::agtype ? '{"id": 1688849860263937, "label": "EDGE", "end_id": 1970324836974593, "start_id": 1407374883553281, "properties": {"n": 100}}::edge';`

Reply:
No, it wont.

```
new=# SELECT '[1,2,3]'::agtype ? '{"id": 1688849860263937, "label": "EDGE", "end_id": 1970324836974593, "start_id": 1407374883553281, "properties": {"n": 100}}::edge'::agtype;
 ?column? 
----------
 f
(1 row)

new=# SELECT '{"id": 281474976710658, "label": "", "properties": {"n": 100}}'::agtype ? '{"id": 1688849860263937, "label": "EDGE", "end_id": 1970324836974593, "start_id": 1407374883553281, "properties": {"n": 100}}::edge';
 ?column? 
----------
 f
(1 row)
```